src/utils.ts

```diff
@@ -85,3 +86,19 @@ export const getEndorctlChecksum = (
       return "";
   }
 };
+
+export const writeJsonToFile = (jsonString: string) => {
+  const { GITHUB_RUN_ID, RUNNER_TEMP } = process.env;
+  const fileName = `result-${GITHUB_RUN_ID}.json`;
+  const uploadPath = path.resolve(RUNNER_TEMP ?? __dirname);
+  const filePath = path.resolve(RUNNER_TEMP ?? __dirname, fileName);
+  fs.writeFile(filePath, jsonString, "utf8", function (err) {
```

Comment:
Should this instead use [`fs/promises`](https://nodejs.org/docs/latest-v16.x/api/fs.html#promise-example), so that the write here may be awaited? Currently, the call to upload the artifact could happen before the file write here finishes.